### ISSUE TITLE
Findings

### ISSUE THREAD

**Issue by sAleksovski** [ytl-RaymondLau/react-native-widget-demo#1]
I tried to reproduce the issue.

Can you explain the `widget.gif` recording? It both works and is not working in it, I don't know what is the difference.

Using the `app-release.apk` from this repo I managed to reproduce the issue, both on emulator and device.

I tried building it locally before I make any changes to the library, and strangely it worked correctly.
I've committed this build as `local.apk`.

To make sure it was not something on my machine, I used eas build to build the app. That build worked correctly too.
I've committed this build as `eas.apk`.

I'm really not sure why your build is not working correctly.
I extracted the apk files, and compared all the files inside.
I'll attach the checksums for each build here, so you can use `diff` to compare them.
I see that `libexpo-modules-core.so` is different in every build, but I don't know why.

Other differences I see are files that have checksums inside them, and some other binaries.
The image differences between local and github apks look the same, so it shouldn't be a problem.

[eas_checksum.txt](https://github.com/ytl-RaymondLau/react-native-widget-demo/files/14657475/eas_checksum.txt)
[github_checksum.txt](https://github.com/ytl-RaymondLau/react-native-widget-demo/files/14657476/github_checksum.txt)
[local_checksum.txt](https://github.com/ytl-RaymondLau/react-native-widget-demo/files/14657477/local_checksum.txt)

Can you try using eas build?

**Comment by ytl-RaymondLau** [comment]
After rebooting my laptop and clearing everything, the issue seems to have resolved itself. However, I'm still uncertain about what caused the problem in the first place.

Regardless, I want to express my gratitude for your assistance.

**Comment by ytl-RaymondLau** [comment]
Hi @sAleksovski, I am encountering the same issue in expo project. Based on my investigation, I have found out that the getBitmap function does not handle the cached file import.

`react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/builder/widget/utils/ResourceUtils.java`

```
public static Bitmap getBitmap(ReactApplicationContext context, String source) throws IOException {
        if (isResource(source)) {
            int resourceId = getResourceId(context, "drawable", source);
            return BitmapFactory.decodeResource(context.getResources(), resourceId);
        } else if (source.startsWith("data:")) {
            String base64String = source.split("base64,")[1];
            byte[] decodedString = Base64.decode(base64String, Base64.DEFAULT);
            return BitmapFactory.decodeByteArray(decodedString, 0, decodedString.length);
        } else {
            InputStream input = getInputStreamFromSource(source);
            return BitmapFactory.decodeStream(input);
        }
    }
```

Example: the source is  "file:///<image>".

I am currently using a patch to fix it. 

```
diff --git a/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/builder/widget/utils/ResourceUtils.java b/android/src/main/java/com/reactnativeandroidwidget/builder/widget/utils/ResourceUtils.java
index 2a66647..ca4980f 100644
--- a/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/builder/widget/utils/ResourceUtils.java
+++ b/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/builder/widget/utils/ResourceUtils.java
@@ -15,6 +15,7 @@ import com.caverock.androidsvg.SVG;
 import com.caverock.androidsvg.SVGParseException;
 import com.facebook.react.bridge.ReactApplicationContext;
 
+import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.HttpURLConnection;
@@ -40,6 +41,8 @@ public class ResourceUtils {
             String base64String = source.split("base64,")[1];
             byte[] decodedString = Base64.decode(base64String, Base64.DEFAULT);
             return BitmapFactory.decodeByteArray(decodedString, 0, decodedString.length);
+        } else if (isFilePath(source)) {
+            return BitmapFactory.decodeFile(source.replaceFirst("file://", ""));
         } else {
             InputStream input = getInputStreamFromSource(source);
             return BitmapFactory.decodeStream(input);
@@ -75,6 +78,10 @@ public class ResourceUtils {
         return uri.getScheme() == null;
     }
 
+    private static boolean isFilePath(String source) {
+        return source.startsWith("file://");
+    }
+
     private static int getResourceId(Context context, String resourceType, String name) {
         if (name == null || name.isEmpty()) {
             return 0;
```

Kindly look into this issue. Thanks in advance.

**Comment by sAleksovski** [comment]
I just released `0.13.1` with your patch, can you try it?

**Comment by ytl-RaymondLau** [comment]
Thank you for the quick update! I've just tried version 0.13.1 with the patch and everything is working perfectly. I appreciate your prompt assistance with this issue.
